src/main/kotlin/it/pagopa/wallet/controllers/WalletNotifyController.kt

```diff
@@ -0,0 +1,27 @@
+package it.pagopa.wallet.controllers
+
+import it.pagopa.generated.npgnotification.api.NotifyApi
+import it.pagopa.generated.npgnotification.model.NotificationRequestDto
+import it.pagopa.wallet.services.WalletService
+import java.util.*
+import lombok.extern.slf4j.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ServerWebExchange
+import reactor.core.publisher.Mono
+
+@RestController
+@Slf4j
+class WalletNotifyController(@Autowired private val walletService: WalletService) : NotifyApi {
+
+    override fun notifyWallet(
+        correlationId: UUID?,
```

Comment:
Well done! 🚀
Not a real issue, but there is no way to enforce the plugin to generate a method signature with non-nullable arguments?

Reply:
This is caused by the used generator `spring`. I would suggest to used the kotlin-spring instead. For the NPG api it was chosen to use spring generator for issue related to oneOf openapi generation